x-pack/plugins/apm/public/components/shared/PropertiesTable/tabConfig.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { get, indexBy, uniq } from 'lodash';
+import { first, has } from 'lodash';
+import { StringMap } from 'x-pack/plugins/apm/typings/common';
+import { APMError } from 'x-pack/plugins/apm/typings/es_schemas/Error';
+import { Transaction } from 'x-pack/plugins/apm/typings/es_schemas/Transaction';
+
+export type TabKey =
+  | keyof Transaction
+  | keyof APMError
+  | 'transaction.custom'
+  | 'error.custom';
+
+export interface Tab {
+  key: TabKey;
+  label: string;
+}
+
+interface TabConfig extends Tab {
+  required: boolean;
+  presortedKeys: string[];
+}
+
+export function getTabsFromObject(obj: Transaction | APMError): Tab[] {
+  return TAB_CONFIG.filter(
+    ({ key, required }) => required || has(obj, key)
+  ).map(({ key, label }) => ({ key, label }));
+}
+
+interface GenericTab {
```

Comment:
This can be called `Tab` if we move the other one to be the more specific `PropertyTab`, `PropertyTabConfig`, `PropertyTabKey` etc

Reply:
I ended up inlining it.